### PR TITLE
bugfix: only list container stats which are created by cri

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### Ⅰ. Describe what this PR did

When CRI manager list container stats, filter the container which is not created by CRI.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

No

### CODE REVIEW

File: cri/v1alpha1/cri.go [hunk 1]
```diff
@@ -822,7 +822,7 @@ func (c *CriManager) ContainerStats(ctx context.Context, r *runtime.ContainerSta
 func (c *CriManager) ListContainerStats(ctx context.Context, r *runtime.ListContainerStatsRequest) (*runtime.ListContainerStatsResponse, error) {
 	opts := &mgr.ContainerListOption{All: true}
 	filter := func(c *mgr.Container) bool {
-		return true
+		return c.Config.Labels[containerTypeLabelKey] == containerTypeLabelContainer
```
Comment (fuweid) [inline cri/v1alpha1/cri.go]: only check labels here?

File: cri/v1alpha1/cri.go [hunk 2]
```diff
@@ -822,6 +822,19 @@ func (c *CriManager) ContainerStats(ctx context.Context, r *runtime.ContainerSta
 func (c *CriManager) ListContainerStats(ctx context.Context, r *runtime.ListContainerStatsRequest) (*runtime.ListContainerStatsResponse, error) {
 	opts := &mgr.ContainerListOption{All: true}
 	filter := func(c *mgr.Container) bool {
+		if c.Config.Labels[containerTypeLabelKey] != containerTypeLabelContainer {
+			return false
+		}
```
Comment (Starnop) [inline cri/v1alpha1/cri.go]: Maybe add a blank here to distinguish between filtering created by CRI Manager and filtering by request and it's beeter to write comments. WDTY?

**Comment by Starnop** [comment]
Except mentioned above, LGTM.

File: cri/v1alpha2/cri_utils.go [hunk 3]
```diff
@@ -814,6 +814,20 @@ func filterCRIContainers(containers []*runtime.Container, filter *runtime.Contai
 	return filtered
 }
 
+// matchLabelSelector returns true if labels cover selector.
+func matchLabelSelector(selector, labels map[string]string) bool {
```
Comment (Starnop) [inline cri/v1alpha2/cri_utils.go]: This function is uncoupled with `runtime`, maybe create a directory like `utils` under the cri directory for more public utility methods and move this funtion over. WDYT?

Comment (YaoZengzeng) [reply cri/v1alpha2/cri_utils.go]: Yes, we could do it in another PR and move all the public utility methods to there.

File: cri/v1alpha1/cri.go (hunk 2 above)
Comment (YaoZengzeng) [reply cri/v1alpha1/cri.go]: Updated

File: cri/v1alpha2/cri_utils.go (hunk 3 above)
Comment (fuweid) [reply cri/v1alpha2/cri_utils.go]: @YaoZengzeng please put it in this PR. 😄 can you do this?

File: cri/v1alpha2/cri.go [hunk 4]
```diff
@@ -918,6 +919,20 @@ func (c *CriManager) ContainerStats(ctx context.Context, r *runtime.ContainerSta
 func (c *CriManager) ListContainerStats(ctx context.Context, r *runtime.ListContainerStatsRequest) (*runtime.ListContainerStatsResponse, error) {
 	opts := &mgr.ContainerListOption{All: true}
```
[the remark this answers is not in the document]
Comment (allencloud) [reply cri/v1alpha2/cri.go]: I think it all depends on what does stats refer to in details.
When using Moby, we can use the API /containers/{id}/stats for a created/stopped container, but I do not think it return non-zero data.

However, if the stats here refer some statistics about the amount of usage in some resources. Even if container is stopped, the usage is still stats of it. So I think it makes sense to support stats API for stopped container.

In a word, it all depends how Kubernetes defines the stats.

**Pull request merged**